tilings/algorithms/minimal_gridded_perms.py

```diff
@@ -485,9 +485,15 @@ def _process_work_packet(
             # if gp was one of the initial_gps that satisfied obs/reqs, but
             # we weren't sure at the time if it was minimal, then now is the
             # time to check and yield
-            if qpacket.gp in initial_gps_to_auto_yield:
+            if curr_len != len(qpacket.gp):
+                curr_len = len(qpacket.gp)
+                for gp in initial_gps_to_auto_yield[curr_len]:
+                    if not yielded_subgridded_perm(qpacket.gp):
+                        yielded.add(gp)
+                        yield gp
+            if qpacket.gp in initial_gps_to_auto_yield[curr_len]:
                 # removing this speed up a later check by a tiny bit
-                initial_gps_to_auto_yield.remove(qpacket.gp)
+                initial_gps_to_auto_yield[curr_len].remove(qpacket.gp)
```

Comment:
this feels unnecessary - you could mark each length as empty after yielding from it

Note, qpacket.gp should never be an initial_gps if you put that continue in the prepare queue method

Reply:
I added the `continue` we discussed and removed the if statement completely.